src/main/java/com/synopsys/integration/blackduck/nexus3/task/common/CommonMetaDataProcessor.java

```diff
@@ -61,43 +63,58 @@ public void setAssetVulnerabilityData(VulnerabilityLevels vulnerabilityLevels, A
     }
 
-    public void addAllAssetVulnerabilityCounts(List<RiskCountView> vulnerabilities, VulnerabilityLevels vulnerabilityLevels) {
-        for (RiskCountView riskCountView : vulnerabilities) {
-            String riskCountType = riskCountView.getCountType().name();
-            int riskCount = riskCountView.getCount();
+    public void addAllAssetVulnerabilityCounts(List<ComponentVersionRiskProfileRiskDataCountsView> vulnerabilities, VulnerabilityLevels vulnerabilityLevels) {
+        for (ComponentVersionRiskProfileRiskDataCountsView riskCountView : vulnerabilities) {
+            ComponentVersionRiskProfileRiskDataCountsCountTypeType riskCountType = riskCountView.getCountType();
+            BigDecimal riskCount = riskCountView.getCount();
             vulnerabilityLevels.addXVulnerabilities(riskCountType, riskCount);
         }
     }
 
-    public void addMaxAssetVulnerabilityCounts(List<RiskCountView> vulnerabilities, VulnerabilityLevels vulnerabilityLevels) {
-        String highestSeverity = "";
-        for (RiskCountView vulnerability : vulnerabilities) {
-            RiskCountType severity = vulnerability.getCountType();
-            int severityCount = vulnerability.getCount();
-            if (RiskCountType.HIGH.equals(severity) && severityCount > 0) {
-                highestSeverity = VulnerabilityLevels.HIGH_VULNERABILITY;
-                break;
-            } else if (RiskCountType.MEDIUM.equals(severity) && severityCount > 0) {
-                highestSeverity = VulnerabilityLevels.MEDIUM_VULNERABILITY;
-            } else if (RiskCountType.LOW.equals(severity) && severityCount > 0) {
-                highestSeverity = VulnerabilityLevels.LOW_VULNERABILITY;
-            }
+    public void addMaxAssetVulnerabilityCounts(List<ComponentVersionRiskProfileRiskDataCountsView> vulnerabilities, VulnerabilityLevels vulnerabilityLevels) {
+        Optional<ComponentVersionRiskProfileRiskDataCountsCountTypeType> highestSeverity = vulnerabilities.stream()
+                                                                                               .filter(Objects::nonNull)
+                                                                                               .filter(this::hasVulnerabilities)
+                                                                                               .map(ComponentVersionRiskProfileRiskDataCountsView::getCountType)
+                                                                                               .max(Comparator.comparingInt(this::getCountTypePriority));
+
+        if (highestSeverity.isPresent()) {
+            vulnerabilityLevels.addVulnerability(highestSeverity.get());
         }
-        if (StringUtils.isNotBlank(highestSeverity)) {
-            vulnerabilityLevels.addVulnerability(highestSeverity);
+    }
+
+    private int getCountTypePriority(ComponentVersionRiskProfileRiskDataCountsCountTypeType countType) {
+        if (null == countType) {
```

Comment:
I don't think this first check is necessary. Equals should be false if countType is null